### DISCUSSION TITLE
is this keeping all records in memory?

### DISCUSSION THREAD

**Discussion by rejetto** [heineiuo/rippledb#168] [Q&A] [answered]
....other than on file, of course

**Comment by heineiuo** [comment]
@rejetto Thanks for your question! This depends which 'ENV' you use, the default ENV is https://github.com/heineiuo/rippledb/blob/5a34bd4eb98f572b57ddcb82d8897e2b2f30f3ae/port/node/index.ts#L17-L60 and it persist records on file system.

**Comment by rejetto** [comment]
well, i don't mean persistency, i want to know the memory footprint at run-time.
If i have data for 1GB, will rippledb use ram for 1GB+?

**Reply by heineiuo** [reply, marked as answer]
No, only records in [MemTable](https://github.com/heineiuo/rippledb/blob/master/src/MemTable.ts) are stored in memory, when some [condition](https://github.com/heineiuo/rippledb/blob/5a34bd4eb98f572b57ddcb82d8897e2b2f30f3ae/src/Database.ts#L729) reached, a new MemTable will be created and the old MemTable will turn into immutable and write records to disk.
